apps/studio/components/interfaces/Database/CronJobs/CreateCronJobSheet.tsx

```diff
@@ -0,0 +1,396 @@
+})
+const sqlSnippetSchema = z.object({
+  type: z.literal('sql_snippet'),
+  snippet: z.string().trim().min(1),
+})
+
+const FormSchema = z.object({
+  name: z.string().trim().min(1, 'Please provide a name for your cron job'),
+  schedule: z
+    .string()
+    .trim()
+    .min(1)
+    .refine((value) => {
+      try {
+        CronToString(value)
+      } catch {
+        return false
+      }
+      return true
+    }, 'The schedule needs to be in a Cron format.'),
+  values: z.discriminatedUnion('type', [
+    edgeFunctionSchema,
+    httpRequestSchema,
+    sqlFunctionSchema,
+    sqlSnippetSchema,
+  ]),
+})
+
+export type CreateCronJobForm = z.infer<typeof FormSchema>
+export type CronJobType = CreateCronJobForm['values']
+
+const FORM_ID = 'create-cron-job-sidepanel'
+
+export const CreateCronJobSheet = ({
+  selectedCronJob,
+  isClosing,
+  setIsClosing,
+  onClose,
+}: CreateCronJobSheetProps) => {
+  const isEditing = !!selectedCronJob?.jobname
+  const [showEnableExtensionModal, setShowEnableExtensionModal] = useState(false)
+  const { mutate: upsertCronJob, isLoading } = useDatabaseCronJobCreateMutation()
+
+  const canToggleExtensions = useCheckPermissions(
+    PermissionAction.TENANT_SQL_ADMIN_WRITE,
+    'extensions'
+  )
+
+  const cronJobValues = parseCronJobCommand(selectedCronJob?.command || '')
+
+  const form = useForm<CreateCronJobForm>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      name: selectedCronJob?.jobname || '',
+      schedule: selectedCronJob?.schedule || '',
+      values: cronJobValues,
+    },
+  })
+
+  const { project } = useProjectContext()
+  const isEdited = form.formState.isDirty
+
+  // if the form hasn't been touched and the user clicked esc or the backdrop, close the sheet
+  if (!isEdited && isClosing) {
+    onClose()
+  }
+
+  const onClosePanel = () => {
+    if (isEdited) {
+      setIsClosing(true)
+    } else {
+      onClose()
+    }
+  }
+
+  const onSubmit: SubmitHandler<CreateCronJobForm> = async ({ name, schedule, values }) => {
+    let command = ''
+    if (values.type === 'edge_function') {
+      command = buildHttpRequestCommand(
+        values.method,
+        values.edgeFunctionName,
+        values.httpHeaders,
+        values.httpParameters,
+        values.timeoutMs
+      )
+    } else if (values.type === 'http_request') {
+      command = buildHttpRequestCommand(
+        values.method,
+        values.endpoint,
+        values.httpHeaders,
+        values.httpParameters,
+        values.timeoutMs
+      )
+    } else if (values.type === 'sql_function') {
+      command = `'CALL ${values.schema}.${values.functionName}()'`
+    } else {
+      command = `$$${values.snippet}$$`
+    }
+
+    const query = buildCronQuery(name, schedule, command)
+
+    upsertCronJob(
+      {
+        projectRef: project!.ref,
+        connectionString: project?.connectionString,
+        query,
+      },
+      {
+        onSuccess: () => {
+          if (isEditing) {
```

Comment:
this need an error toast?

Reply:
no, it's to decide whether the panel is used for editing existing cron job or creating a new one.